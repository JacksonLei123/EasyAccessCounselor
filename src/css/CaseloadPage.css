.square {
    width: 200px;
    height: 200px;
    border-radius: 10px;
    margin: 10px;
    overflow: hidden;
    float: left;
    text-align: left;
}

#root {
    width: auto;
    margin: auto;
}

#grid {
    text-align: center;
    width: 880px;
    margin: auto
}

.card {
    background-color: #ACE7FF!important;
    height: 200px!important;
}

.card:hover{
    background-color: rgba(172, 232, 255, 0.753)!important;
    cursor: pointer;
}

.blockview-child {
    width: 80%;
    height: 200px;
    border-radius: 10px;
    margin: 20px auto;
    overflow: hidden;
}

span.clear { 
    clear: left; 
    display: block; 
}