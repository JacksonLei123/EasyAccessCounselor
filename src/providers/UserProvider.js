import React, { Component, createContext } from "react";
import { db, auth } from '../firebase/firebase.js';
import firebase from 'firebase/app';

export const UserContext = createContext({ user: null, selectedCohort: null, cohorts: [] });

class UserProvider extends Component {
    state = {
        user: null,
        selectedCohort: null,
        cohorts: []
    };

    userUpdate = (user) => {
        if (user) {
            db.collection("counselors").doc(user.uid).get()
            .then(querySnapshot=> {
                return querySnapshot.data().cohorts;
                })
            .then(cohorts=>{
                db.collection("student_counselors").where(firebase.firestore.FieldPath.documentId(), "in", cohorts).get()
                .then(querySnapshot=>{
                    return querySnapshot.docs.map(doc =>{
                        var new_doc = doc.data();
                        new_doc.id = doc.id;
                        return new_doc;
                      });
                })
                .then(data=>this.setState({
                    user: user,
                    cohorts: data,
                    selectedCohort: this.state.selectedCohort ? this.state.selectedCohort : data.length > 0 ?  data[0].id : null
                  }))
                  .catch(error=>{console.log(error)});
            })
            .catch(error=>{console.log(error)});
          } else {
              this.setState({
                user: null,
                selectedCohort: null,
                cohorts: []
            });
          }
    }

    changeSelectedCohort = (e) =>{
        this.setState({selectedCohort: e.target.value});
    }

    addCohort = (cohortName, cohortID) => {
        var new_cohorts = this.state.cohorts.slice();
        new_cohorts.push({name: cohortName, id: cohortID});
        this.setState({cohorts: new_cohorts, selectedCohort: cohortID});
    }

    componentDidMount = () => {
        this._isMounted = true;
        auth.onAuthStateChanged(userAuth => {
            this._isMounted && this.userUpdate(userAuth);
        })
    }

    componentWillUnmount = () => {
        this._isMounted = false;
    }



    render() {
        return(
            <UserContext.Provider value={{state:this.state, addCohort: this.addCohort, changeSelectedCohort: this.changeSelectedCohort}}>
                {this.props.children}
            </UserContext.Provider>
        )
    }
}

export default UserProvider;