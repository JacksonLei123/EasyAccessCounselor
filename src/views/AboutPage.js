import React from 'react';
import { db } from '../firebase/firebase';

class AboutPage extends React.Component {
    state = {
        data : null,
    }

  async componentDidMount() {
    let query = await db.collection("students").where("cuid", "==", this.props.cuid).get();
    this.setState({data: query.docs.map(doc => doc.data())})
  }

  displayTable() {
    let students = [];
    this.state.data.forEach(student => students.push(student));
    let table = 
        <div>
          {students.map(student => <div key={student.name}>
            {student.firstName} {student.lastName}
          </div>)}
        </div>
    return table;
  }

  render() {
      return (
        <div>
          This is just a tinkering page for right now. This currently has an example of how to
          get the info for a counselor's students.
          <h2>My Students</h2>
          {this.state.data ? this.displayTable() : ""}
        </div>
      );
  }
}

export default AboutPage;