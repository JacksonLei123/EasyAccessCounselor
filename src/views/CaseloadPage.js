import React from 'react';
import '../css/searchBar.css';
import '../css/CaseloadPage.css';

class CaseloadPage extends React.Component {
  constructor(props){
    super(props);
    this.state = ({
      searchString: ""
    });
    this.fields = [
      "Goal",
      "Meetings",
      "Tasks",
      "Visit",
      "Testing",
      "List",
      "Fee Waiver",
      "Early Apps",
      "Essay",
      "Teacher Recs",
      "Counselor Rec",
      "Resume",
      "FAFSA",
      "CSS Profile",
      "Results",
      "Verification",
      "Award Letters",
      "Appeal",
      "Spring Visit",
      "Decision",
      "Latest"
    ];
    
  }

  getContent = (e) => {
    console.log(e.target.value);
  }
  render(){
    console.log("rendered");
    return (
      <div className="caseload-content">
        <div className="profiles-header">
          <input type="text" id="myInput" onKeyUp={this.changeSearchString} placeholder="Search for Students.." />
        </div>
      </div>
    );
  }
}

export default CaseloadPage;

/*

<table id="caseload-table">
            <thead>
              {this.fields.map(field => {
                return <th className="caseload-headcell">{field}</th>;
              })}

            </thead>
            <tbody>
            <tr>
              <td className="tableCell" ><input className="inputCell" type="text" defaultValue="Hi" onChange={this.getContent}/></td>
            </tr>
            </tbody>
          </table>

*/