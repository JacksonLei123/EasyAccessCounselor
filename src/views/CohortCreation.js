import React from 'react';
import '../css/CohortCreation.css';

class FileUploadSelection extends React.Component {
    constructor(props){
        super(props);
        this.state = {fileName: null, file: null}
    }

    preventDefault = e => e.preventDefault();

    componentDidMount() {
        window.addEventListener("dragover", this.preventDefault);
        window.addEventListener("drop", this.preventDefault);
    }

    componentWillUnmount() {
        window.removeEventListener("dragover", this.preventDefault);
        window.removeEventListener("drop", this.preventDefault);
    }

    handleFile = (file) => {
        let fileType = file.name.split('.').pop();
        if (!["xls", "xlsv", "csv"].includes(fileType)){
            alert('File must be .xls, .xlsv, or .csv');
        } else {
            this.setState({fileName: file.name, file: file});
        }
    }

    onChange = (e) => {
        this.handleFile(e.target.files[0]);
    }

    handleDrop = (e) => {
        if (e.dataTransfer.files && e.dataTransfer.files.length > 0) {
            this.handleFile(e.dataTransfer.files[0]);
        }
    }

    onSubmit = () => {
        if (this.state.file){
            let fileReader = new FileReader();
            console.log("reading");
            fileReader.addEventListener('load', (event)=> {
                console.log(event.target.result);
                this.props.setData(event.target.result);
                this.props.changePanel("fieldMatching");
            });
            fileReader.readAsText(this.state.file);
        }
    }
    render(){
        return (
            <div className="file-upload-wrapper">
                <p>Upload your student data.</p>
                <div className="uploadbox" onDrop={this.handleDrop}>{this.state.fileName ? this.state.fileName : "Drag and drop file."}</div>
                <div className="file-upload-buttons">
                    <input className="uploadinput" id="file" type="file" name="file" accept=".csv, .xlsx, .xls" onChange={this.onChange} />
                    <label htmlFor="file" className="select-entry-btn">Browse</label>
                    <button className={"select-entry-btn" + (this.state.file ? "": " inactive")} onClick={this.onSubmit}>Upload</button>
                </div>
            </div>
        );
    }
}

class DataEntrySelection extends React.Component {
    constructor(props){
        super(props);
        this.state = {selectedState: null}
    }
    onChange = (e)=> {
        this.setState({selectedState: e.target.value});
    }

    onSubmit = ()=> {
        switch(this.state.selectedState){
            case "manual":
                window.location.href = window.location.origin + "/caseload_management";
                break;
            case "upload":
                this.props.changePanel("fileUpload");
                break;
            default:
                console.log("None selected!");
        }

    }
    
    render() {
        return (
        <div className="select-entry-type">
            <p>Data Entry</p>
            <div className="rad-select-group">
                    <input className="rad-select-btn" type="radio" name="dataEntry" value="manual" onChange={this.onChange}/>
                    <p>Manually enter student data</p>
            </div>  
            <div className="rad-select-group">
                    <input className="rad-select-btn" type="radio" name="dataEntry" value="upload" onChange={this.onChange}/>
                    <p>Upload data from your computer</p>
            </div>  
            <button className={"select-entry-btn" + (this.state.selectedState ? "": " inactive")} onClick={this.onSubmit}>Create</button>
        </div>);
    }
}

class FieldMatches extends React.Component {
    constructor(props){
        super(props);
        let fieldsMap = new Map();
        this.props.data.split('\n')[0].split(',').forEach(field=>fieldsMap.set(field, "None"));
        this.state = {
            fieldsMap: fieldsMap
        };
        this.fields = {
        "UniqueID": "id",
        "First Name": "firstName",
        "Last Name": "lastName",
        "Date of Birth": "dob",
        "Ethnicity": "ethnicity",
        "Gender": "gender",
        "GPA": "gpa",
        "Class Rank": "classRank",
        "SAT": "sat",
        "ACT": "act",
        "Student Email": "studentEmail",
        "Student Email 2": "studentEmail2",
        "Student Phone": "studentPhone",
        "Parent Email": "parentEmail",
        "Parent Email 2": "parentEmail2",
        "Parent Phone": "parentPhone",
        "Parent Phone 2": "parentPhone2",
        "Student Address": "studentAddress",
        "Total Meetings": "totalMeetings",
        "Individual Meetings": "individualMeetings",
        "Group Meetings": "groupMeetings",
        "Event Meetings": "eventMeetings",
        "State": "state",
        "Zipcode": "zipcode",
        "EFC": "efc",
        "Field of Study/Major 1": "major",
        "Field of Study/Major 2": "major2",
        "Safety Colleges": "safetyColleges",
        "Target Colleges": "targetColleges",
        "Reach Colleges": "reachColleges",
        "Counselor Additions": "additions",
        "Want to Attend (Region)": "region",
        "College Size": "collegeSize",
        "College Setting": "collegeSetting",
        "College Diversity (% URM)": "collegeDiversity",
        "College Diversity (Types)": "collegeDiversityTypes",
        "Religion": "religion",
        "Military/ROTC": "rotc",
        "Athletics": "athletics"
    }
    }

    selectField = (e, field) => {
        let new_fields = new Map(this.state.fieldsMap);
        new_fields.set(field[0], e.target.value);
        this.setState({fieldsMap: new_fields});

    }

    render(){
        return (
            <div>
                <p>Match your upload's fields with the fields in the Easy Access database, or create a new field</p>
                <div className="field-titles">
                    <p>Upload's Field</p>
                    <p>Easy Access Field</p>
                </div>
                <div className="fields-view">
                    {Array.from(this.state.fieldsMap).map(field=>{
                        return(
                        <div key={field[0]} className="field-select">
                            <p>{field[0]}</p>
                            <select value={field[1]} onChange={e=>this.selectField(e, field)}>
                                <option value="None">Select Option</option>
                                {Object.entries(this.fields).map(val=>{
                                    return <option key={val[1] + field} value={val[1]}>{val[0]}</option>
                                })}
                            </select>
                        </div>
                        );
                    })}
                </div>        
            </div>
        );
    }
}

class CohortCreation extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            selectedPanel: "dataEntry",
            data: null 
        };
        this.panels = {
            "dataEntry": <DataEntrySelection changePanel={this.changePanel}/>,
            "fileUpload": <FileUploadSelection changePanel={this.changePanel} setData={this.setData}/>,
            "fieldMatching": <FieldMatches data={this.state.data} changePanel={this.changePanel}/>
        }
    }

    changePanel = (panel) => {
        this.setState({selectedPanel: panel});
    }
    
    setData = data => {this.setState({data: data}); this.panels.fieldMatching=<FieldMatches data={data} changePanel={this.changePanel}/>};

    render(){
        return (
            <div className="cohort-creation-content">
                {this.panels[this.state.selectedPanel]}
            </div>
        );
    }
}

export default CohortCreation;