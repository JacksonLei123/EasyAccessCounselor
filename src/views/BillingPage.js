import React, { useContext } from "react";
import { UserContext } from '../providers/UserProvider';
import * as firebase from 'firebase/app';
import { auth, db } from "../firebase/firebase";
import '../css/BillingPage.css';

class BillingPage extends React.Component {

    static contextType = UserContext;
    
    constructor(props) {
        super(props);
        this.state = ({
            userEmail: '',
            cohortName: ''
        });
    }
    /*this.context.state.cohorts.find(o => o.uid == this.context.state.selectedCohort)
    getUserData = () => {
        this.setState({
            userEmail: ,
            cohortName: this.context.state.cohorts.find(o => o.uid == this.context.state.selectedCohort)
        });
    }
    componentDidMount() {
        this.getUserData();
    }*/

    editBillingInfo = () => {
        // if (this.context.state !== null && window.confirm("Are you sure you want to delete this cohort?")){
        //     db.collection("student_counselors").doc(this.context.state.selectedCohort).delete()
        //     .then(result=>this.context.deleteCohort(this.context.state.selectedCohort));
        
        // }
    }


    render() {
        return (
            <div id="billingContainer">
                <h1 id="billingHeader">Billing Information</h1>
                <div class="formBlock">
                    <div class="billingRow">
                        <h3 className="rowPrefix">Cardholder Name</h3>
                        <input class="inputLabel" id="email" type="text" placeholder={this.context.state !== null ? this.context.state.user.email : ""} />
                        {/* insert stripe component here */}
                    </div>
                    <div class="billingRow">
                        <h3 className="rowPrefix">Card Number</h3>
                        {/* insert stripe component here */}
                    </div>
                    <div class="billingRow">
                        <h3 className="rowPrefix">Expiration Date</h3>
                        {/* insert stripe component here */}
                    </div>
                    <div class="billingRow">
                        <h3 className="rowPrefix">Security Code</h3>
                        {/* insert stripe component here */}
                    </div>
                </div>
                <div class="formBlock">
                    <div class="billingRow">
                        <h3 className="rowPrefix">Billing Address</h3>
                        {/* insert stripe component here */}
                    </div>
                    <div class="billingRow">
                        <h3 className="rowPrefix">City</h3>
                        {/* insert stripe component here */}
                    </div>
                    <div class="billingRow">
                        <h3 className="rowPrefix">State</h3>
                        {/* insert stripe component here */}
                    </div>
                    <div class="billingRow">
                        <h3 className="rowPrefix">Zipcode</h3>
                        {/* insert stripe component here */}
                    </div>
                </div>
                <button id="editBilling" onClick={()=>this.editBillingInfo()}>Edit Billing Information</button>
            </div>
        );
    }

    
}
export default BillingPage;