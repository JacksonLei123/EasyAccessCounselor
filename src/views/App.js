import React, { useContext, useState, createContext } from 'react';

import { 
  Router,
  Switch,
  Route
} from "react-router-dom";

import { UserContext } from "../providers/UserProvider";
import { createBrowserHistory } from 'history';

import '../css/App.css';
import Login from './LoginPage';
import HomeDefault from './HomePageDefault';
import Dashboard from './Dashboard';
import Account from './AccountPage';
import Caseload from './CaseloadPage';
import CollegeList from './CollegeListPage';
import Settings from './SettingsPage';
import CohortCreation from './CohortCreation';
import Notes from './NotesPage';
import StudentProfilesPage from './StudentProfilesPage';
import Header from '../components/Header';
import Sidenav from '../components/Sidenav';

export const history = createBrowserHistory();

export default function App() {
  const user = useContext(UserContext);
  const isLoggedIn = user.state.user ? 'true' : '';
  return (
    <Router history={history}>
      <div className="page-container">
        <Sidenav isLoggedIn={isLoggedIn}/>
        <div className="page-content">
          <Switch>
            <Route path="/profiles">
              <Header isLoggedIn={isLoggedIn}/>
              { user.state.user ? <StudentProfilesPage /> : <Login /> }
            </Route>
            <Route path="/caseload_management">
              <Header isLoggedIn={isLoggedIn}/>
              { user.state.user ? <Caseload cuid={user.uid}/> : <Login /> }
            </Route>
            <Route path="/settings">
              <Header isLoggedIn={isLoggedIn}/>
              { user.state.user ? <Settings currentUser={user} /> : <Login /> }
            </Route>
            <Route path="/cohortcreation">
              <Header isLoggedIn={isLoggedIn}/>
              { user.state.user ? <CohortCreation /> : <Login /> }
            </Route>
            <Route path="/cohortcreation2">
              <Header isLoggedIn={isLoggedIn}/>
              { user.state.user ? <CohortCreation /> : <Login /> }
            </Route>
            <Route path="/">
              <Header isLoggedIn={isLoggedIn}/>
              { user.state.user ? <Dashboard cuid={user.uid} /> : <HomeDefault /> }
            </Route>
          </Switch>
        </div>
      </div>
    </Router>
  );
}