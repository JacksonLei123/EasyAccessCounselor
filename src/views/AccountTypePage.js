import React, {useState} from 'react';
import '../css/AccountTypePage.css';
import {UserContext} from '../providers/UserProvider';

import {auth} from '../firebase/firebase';
import {history} from './App';

class AccountTypeComponent extends React.Component {
    static contextType = UserContext;
    constructor(props){
        super(props);
        this.state = ({
        });

        this.content = {
            name: props.content.name,
            type: props.content.type,
            header: props.content.header,
            price: props.content.price,
            desc: props.content.desc,
            list: props.content.list,
            primaryColor: props.content.primaryColor
        }
        
    }

    render() {
        return(
            <div className="typeContainer">
                <div className="typeHeader">
                    <h3>{this.content.header}</h3>
                </div>
                <div className="typeMainBody">
                    <h5>Always</h5>
                    <div className="typeCostContainer">
                    <h1 className="typeCost">{this.content.price}</h1><h4 className="typeMonth">/month</h4>
                    </div>
                    <p className="typeDesc">{this.content.desc}</p>
                    <ul className="typeList">{this.content.list}</ul>
                    <button className="typeButton" onClick={()=>setType(this.content.type)}>Sign-up for {this.content.name}</button>
                </div>
            </div>
        )
    }
}

function setType(type) {
    switch(type) {
        case 'free':
            // TODO: set user type to free
            history.push("/cohortcreation");
            if (document.getElementsByClassName("sidenav-container").length === 1) {
                document.getElementsByClassName("sidenav-container")[0].style.display = "initial";
            }

            if (document.getElementsByClassName("header-select").length === 1) {
                document.getElementsByClassName("header-select")[0].style.display = "initial";
            }
            break;
        case 'counselor':
            // TODO: set user type to counselor
            history.push("/PaymentPortal");
            break;
        case 'school':
            // TODO: set user type to schoo
            history.push("/PaymentPortal");
            break;
        default:
            // TODO: Figure out what should go here
            break;
    }
}

function AccountTypePage() {

    if (document.getElementsByClassName("sidenav-container").length === 1) {
        document.getElementsByClassName("sidenav-container")[0].style.display = "none";
    }
    
    if (document.getElementsByClassName("header-select").length === 1) {
        document.getElementsByClassName("header-select")[0].style.display = "none";
    }

    const freeProps = {
        name: "Free Version",
        type: "free",
        header: "Free Forever",
        price: 0,
        desc: "Save time in your caseload management, organize cohorts",
        list: ["Cohort Creation", "Caseload Management", "Space Holder"],
        primaryColor: [55, 55, 55] // set this to grey this is just a place holder
    }

    const counselorProps = {
        name: "Counselor Plan",
        type: "counselor",
        header: "For Counselors",
        price: 100,
        desc: "Save time and improve your ability to help your students achieve their dreams",
        list: [
            "Premium Cohort Creation",
            "Premium Caseload Management",
            "Space Holder...",
            "Space Holder 2...",
        ],
        primaryColor: [0, 0, 255] // Should be Easy Access blue
    }

    const schoolProps = {
        name: "School Plan",
        type: "school",
        header: "For Schools",
        price: 250,
        desc: "Teamp up on caseload management by granting access to all of your cohorts across your team",
        list: [
            "Multiple Cohort Creation",
            "Team Based Caseload Management",
            "Space Holder...",
            "Space Holder 2..."
        ],
        primaryColor: [255, 100, 0] // Should be orange
    }

    return(
        // login header
        // some kind of body
        <div className="accountTypeOverlay">
            <AccountTypeComponent content = {freeProps}></AccountTypeComponent>
            <AccountTypeComponent content = {counselorProps}></AccountTypeComponent>
            <AccountTypeComponent content = {schoolProps}></AccountTypeComponent>
        </div>
    )
}

export default AccountTypePage;