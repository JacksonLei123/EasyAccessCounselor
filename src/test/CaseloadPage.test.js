import React, { useContext, useState, createContext } from 'react';
import { expectation } from "sinon";
import CaseloadPage from "../views/CaseloadPage";
import UserProvider, { UserContext } from '../providers/UserProvider';
import { DownloadPopup } from "../views/CaseloadPage"
import customHeader from '../components/caseloadPage/customHeader.jsx'
import { auth } from "../firebase/firebase";
import firebase from 'firebase/app'
import Enzyme, { shallow, render, mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";

/* TODO:
    (1) Write a test to make sure that the download popup renders 
    (2) Write a test to make sure that the download popup columns match those in the grid
    (3) Write a test to make sure that the header component renders correctly 
    (?) Test to make sure that newly appended column available in the download popup
*/

Enzyme.configure({ adapter: new Adapter() });
// Sign in firebase user for testing CaseloadPage functionality 
let userState;
function setUserContext(user) {
    userState = { user: user, selectedCohort: null, cohorts: [] };
    console.log('Test User Context Set');
}

beforeAll(async () => {
    jest.setTimeout(10000);
    await auth.signInWithEmailAndPassword("ian.dershem@gmail.com", "kenan116");
    await auth.onAuthStateChanged((user) => {
        setUserContext(user);
    });
});

// Test Shallow Render of Caseload Page
// TODO: Ensure that CaseloadPage actually has access to UserProvider data
test("CaseloadPage shallow render correct", () => {
    // user 
    // shallow() esentially uses a less complicated DOM
    const wrapper = shallow (
        <UserContext.Provider value={{state:userState}}>
            <CaseloadPage/>
        </UserContext.Provider>  
    );
    expect(wrapper).toMatchSnapshot();
});