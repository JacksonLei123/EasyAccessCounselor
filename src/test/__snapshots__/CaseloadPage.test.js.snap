// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaseloadPage shallow render correct 1`] = `ReactWrapper {}`;
